cfme/intelligence/chargeback/rates.py

```diff
@@ -73,14 +77,51 @@ def sub_row_id(self):
                 parent=self)
             return int(attr.rsplit('_', 1)[-1])
 
-        per_time = Select(id=ParametrizedString('per_time_{@row_id}'))
-        per_unit = Select(id=ParametrizedString('per_unit_{@row_id}'))
-        start = Input(id=ParametrizedString('start_{@row_id}_{@sub_row_id}'))
-        finish = Input(id=ParametrizedString('finish_{@row_id}_{@sub_row_id}'))
-        fixed_rate = Input(id=ParametrizedString('fixed_rate_{@row_id}_{@sub_row_id}'))
-        variable_rate = Input(id=ParametrizedString('variable_rate_{@row_id}_{@sub_row_id}'))
-        action_add = Button(title='Add a new tier')
-        action_delete = Button(title='Remove the tier')
+        @ParametrizedView.nested
+        class sub_rows(ParametrizedView):  # noqa
+            PARAMETERS = ('index',)
+            ROOT = ParametrizedLocator(
+                '../tr[./td/input[starts-with(@id, "fixed_rate_{@row_id}_{index}")]]')
+
+            def __locator__(self):
+                return self.browser.element(self.ROOT, parent=self.parent)
+
+            @cached_property
+            def row_id(self):
+                return self.parent.row_id
+
+            def fill(self, values):
+                if not self.is_displayed:
+                    self.__class__(index=self.context['index'] - 1).action_add.click()
```

Comment:
Well, this obviously looks ugly. I won't block, however, on merging, because I believe something like this should also be handled nicely within widgetastic itself. So if we can write down some use cases for this, I can add some functionality into widgetastic for this.